subsys/suit/validator/include/suit_validator.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ * @brief Validator functions for SoC-dependent properties
+ *
+ * @details This module implements functions for validating properties, such as addresses.
+ * 	        The result of validating a property can be different depending on the SoC.
+ *
+ */
+
+#ifndef SUIT_VALIDATOR_H__
+#define SUIT_VALIDATOR_H__
+
+#include <stdint.h>
+#include <stddef.h>
+#include <suit_plat_err.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Validate the location of the update candidate.
+ *
+ * @retval SUIT_PLAT_SUCCESS     The location is valid.
+ * @retval SUIT_PLAT_ERR_ACCESS  The location is forbidden.
+ */
+suit_plat_err_t suit_validator_validate_update_candidate_location(const uint8_t *address,
```

Comment:
Looking at function names: 

suit_validator_validate_update_candidate_location, suit_validator_validate_dfu_partition_location - I am to some extent confused. 

We have two kind of regions to distinguish:
1. an area holding a 'primary' envelope (one containing candidate manifest(s)) - that one must be placed in MRAM
2. area(s) with dfu cache entries - those may be placed both on MRAM and external flash (actually- it can be any kind of non-volatile memory represented as 'external flash')

So what is a mapping between discussed functions and regions listed above?

Perhaps we could consider bettter-describing function names?

Reply:
My mistake, I have skipped the word "cache" by accident. Changed the name of the second function to `suit_validator_validate_cache_pool_location` now it should be more clear and consistent.